### DISCUSSION TITLE
Adding one more tab for crate's features

### DISCUSSION THREAD

**Discussion by HosMercury** [rust-lang/crates.io#8042] [Feature Requests]
I suggest adding one more tab for each crate to doc their features

after versions tab or beside !!

Right now there is no good way to find every crate feature and their usage. Organizing features will facilitate the finding and usefulness of a crate feature and how to use it.

**Comment by LawnGnome** [comment]
This sort of exists today, in two forms:

1. docs.rs includes feature flags for any crate published since late 2020; eg https://docs.rs/crate/tokio/latest/features
2. crates.io shows a bare list of feature flags for a published version; eg https://crates.io/crates/tokio/versions (mouse over the "_n_ features" label in a version row)

We also have #465, which is still open and covers similar ground.

My feeling is that it's generally more appropriate for docs.rs to document feature flags; adding any additional tabs to the crate page is only likely to be confusing since feature flags can change between versions, and the crates.io UI isn't super clear at distinguishing when you're viewing information on a particular crate version or just the "latest" crate version.

But I'm interested in @Turbo87's thoughts on this as well, since he added the second feature listed above a while back.

**Reply by Turbo87** [reply]
the main problem is that there is no official way to document features. similarly, there is no way to document that a feature is an internal/private feature. there is also no way to document whether an optional dependency should be treated as a feature.

due to all of that it is recommended that crates that want to document their features do so either in the doc comments of the crate, or in their README (or both). the former is rendered by docs.rs, the latter by crates.io.

**Reply by HosMercury** [reply]
Thanks for the clarification and the links.
Features docs may need adding some more docs clarification.
